internal/telemetry/Event.go

```diff
@@ -0,0 +1,80 @@
+package telemetry
+
+import (
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+// Event is a telemetry event
+type Event struct {
+	ID     string
+	Type   EventType
+	UserID string
+	Time   time.Time
+	Data   map[DataKey]interface{}
+}
+
+var (
```

Comment:
friendly reminder to use `lower_case_names.go` for file names here

---

I see what youre accomplishing here, but i'd like to avoid this style of private configuration fields.  I think doing so ultimately it pushes us in a direction where we may want to tweak the `Track` interface signature, or introduce some sort of `EventSource` that also sits on the `commandFactory`.  we can chat offline about this one, id like to hear your thoughts and/or any other suggestions you might have

Reply:
I'm not sure why the file names haven't changed.  They are changed on my computer, but not in the git diff so it should have been pushed as lowercase by now.  IDK what is going on with git. maybe it's case insensitive???